### PR TITLE
Allow client logs and messages instrumentation

### LINKED ISSUE #2019: portal: connlib log helpers
- [ ] [Generate signed URLs](https://cloud.google.com/storage/docs/access-control/signing-urls-manually) for POSTing log files 
- [ ] `log_url` message type for clients to request a new POST url
- [ ] terraform HCL for related GCP object storage buckets

### PR DESCRIPTION
**Pull request by AndrewDryga**
Closes #2019

### CODE REVIEW

File: elixir/apps/api/lib/api/relay/socket.ex [hunk 1]
```diff
@@ -11,34 +12,41 @@ defmodule API.Relay.Socket do
...
-
-      {:ok, socket}
-    else
-      {:error, :invalid_token} ->
-        {:error, :invalid_token}
-
-      {:error, reason} ->
-        Logger.debug("Error connecting relay websocket: #{inspect(reason)}")
-        {:error, reason}
+    :otel_propagator_text_map.extract(connect_info.trace_context_headers)
```
Comment (AndrewDryga) [inline elixir/apps/api/lib/api/relay/socket.ex:15]: @thomaseizinger @conectado FYI: if you decide to send W3C trace context headers - I will use the trace you started as a parent.

File: elixir/apps/api/lib/api/client/channel.ex [hunk 2]
```diff
@@ -76,35 +97,56 @@ defmodule API.Client.Channel do
...
 
   def handle_info({:resource_removed, resource_id}, socket) do
+    OpenTelemetry.Tracer.set_current_span(socket.assigns.opentelemetry_ctx)
+    OpenTelemetry.Tracer.add_event("resource_updated", %{resource_id: resource_id})
+
     push(socket, "resource_removed", resource_id)
     {:noreply, socket}
   end
 
+  def handle_in("create_log_sink", _attrs, socket) do
```
Comment (AndrewDryga) [inline elixir/apps/api/lib/api/client/channel.ex:107]: @thomaseizinger this is the message that you need to send from a client to get a signed URL where logs can be uploaded.

**Comment by AndrewDryga** [comment]
Merged to test the Cloud Trace integration on staging.

File: elixir/apps/api/lib/api/relay/socket.ex (hunk 1 above)
Comment (thomaseizinger) [reply elixir/apps/api/lib/api/relay/socket.ex:15]: I'll see to integrate that!

**Pull request merged**